x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/edit_output_flyout/advanced_options_section.tsx

```diff
@@ -0,0 +1,303 @@
+            <EuiFlexItem>
+              <EuiFieldNumber {...diskQueueMaxSizeInput.props} placeholder="Bytes" min={0} />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.diskQueueMaxSizeDescription"
+                  defaultMessage="Limits the disk size for spooling of data. If set too low, data may be lost when agent can't write data to the destination."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+
+        <EuiHorizontalRule />
+
+        <EuiFormRow fullWidth {...loadBalanceEnabledInput.formRowProps}>
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiSwitch
+                data-test-subj="editOutputFlyout.loadBalancingSwitch"
+                {...loadBalanceEnabledInput.props}
+                label={
+                  <FormattedMessage
+                    id="xpack.fleet.settings.editOutputFlyout.loadBalancingSwitchLabel"
+                    defaultMessage="Load Balancing"
+                  />
+                }
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.loadBalancingDescription"
+                  defaultMessage="Once enabled, the agents will balance the load across all the hosts defined for this output. This will increase the number of connections opened by the agent."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+
+        <EuiHorizontalRule />
+
+        <EuiFormRow fullWidth>
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiSwitch
+                data-test-subj="editOutputFlyout.compressionSwitch"
+                {...diskQueueCompressionEnabled.props}
+                label={
+                  <FormattedMessage
+                    id="xpack.fleet.settings.editOutputFlyout.compressionSwitchLabel"
+                    defaultMessage="Compression"
+                  />
+                }
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiSelect
+                data-test-subj="editOutputFlyout.compressionLevelSelect"
+                id="selectCompressionLevel"
+                aria-label="Use aria labels when no actual label is in use"
+                {...compressionLevelInput.props}
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.compressionSwitchDescription"
+                  defaultMessage="Level 1 compression is the fastest, Level 9 however would provide the best compression."
```

Comment:
It's not clear if this compression is applied before putting data in the queue or only before sending data to the output. How should we address this?

I also think "best" is not specific enough.

```suggestion
                  defaultMessage="Level 1 compression is the fastest, Level 9 however would provide the most compression."
```

Reply:
@criamico this suggestion can also be committed.